x-pack/heartbeat/security_linux.go

```diff
@@ -0,0 +1,222 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"runtime"
+	"strconv"
+	"syscall"
+
+	"github.com/elastic/beats/v7/libbeat/common/seccomp"
+	"kernel.org/pub/linux/libs/security/libcap/cap"
+)
+
+func init() {
+	// Here we set a bunch of linux specific security stuff.
+	// In the context of a container, where users frequently run as root, we follow BEAT_SETUID_AS to setuid/gid
+	// and add capabilities to make this actually run as a regular user. This also helps node in synthetics, which
+	// does not want to run as root. It's also just generally more secure.
+	if localUserName := os.Getenv("BEAT_SETUID_AS"); localUserName != "" && syscall.Geteuid() == 0 {
+		err := changeUser(localUserName)
+		if err != nil {
+			panic(err)
+		}
+		err = setCapabilities()
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	switch runtime.GOARCH {
+	case "amd64", "386":
+		err := setSeccompRules()
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+func changeUser(localUserName string) error {
+	localUser, err := user.Lookup(localUserName)
+	if err != nil {
+		return fmt.Errorf("could not lookup '%s': %s", localUser, err)
+	}
+	localUserUid, err := strconv.Atoi(localUser.Uid)
+	if err != nil {
+		return fmt.Errorf("could not parse UID '%s' as int: %s", localUser.Uid, err)
+	}
+	localUserGid, err := strconv.Atoi(localUser.Gid)
+	if err != nil {
+		return fmt.Errorf("could not parse GID '%s' as int: %s", localUser.Uid, err)
+	}
+	// We include the root group because the docker image contains many directories (data,logs)
+	// that are owned by root:root with 0775 perms. The heartbeat user is in both groups
+	// in the container, but we need to repeat that here.
+	err = syscall.Setgroups([]int{localUserUid, 0})
+	if err != nil {
+		return fmt.Errorf("could not prsetgroups: %s", err)
+	}
+
+	// Set the main group as 1000 so new files created are owned by the user's group
+	err = syscall.Setgid(1000)
```

Comment:
Why is this hard-coded to `1000`? Should it be `localUserGid`.

Reply:
Good catch, will fix